bolides/bdf.py

```diff
@@ -927,43 +927,66 @@ def plot_dates(self, freq='1D', logscale=False,
-        """Pull light curve data from neo-bolide.ndc.nasa.gov.
+        """Pull light curve and integrated energy data from neo-bolide.ndc.nasa.gov.
 
         Downloads light curve data from neo-bolide.ndc.nasa.gov, placing it
         into the BolideDataFrame as a column of `~lightkurve.LightCurveCollection` objects.
-        The column is named ``'lightcurves'``
+        The column is named ``'lightcurves'``. Integrated energy data is also downloaded
+        for different satellites and placed into columns starting with ``'energy_'``.
+        This method will only work with BolideDataFrames having an ``_id`` column containing
+        IDs from GLM bolide detections at neo-bolide.ndc.nasa.gov.
 
         Parameters
         ----------
         ids : list of str
-            Optional list of strings representing the bolide ID's that light curves are wanted for.
-            If not used, a light curve is added for every bolide in the BolideDataFrame.
+            Optional list of strings representing the bolide ID's that additional data is
+            needed for. If not used, data is added for every bolide in the BolideDataFrame.
         """
 
-        lclist = []
-        for num, row in tqdm(self.iterrows(), "Downloading data", total=len(self)):  # for each bolide
-
-            # if a subset of ids was specified that excludes this row, skip.
-            if ids is not None and row['_id'] not in ids:
-                continue
-
-            # pull data from website
-            data = requests.get(API_ENDPOINT_EVENT + row['_id']).json()['data'][0]['attachments']
-            row_lcs = []
-
-            # create a LightCurve for each attachment in the data
-            for attachment in data:
-                geodata = attachment['geoData']
-                flux = [point['energy'] for point in geodata]
-                time = [point['time']/1000 for point in geodata]
-                from astropy.time import Time
-                time_obj = Time(time, format='unix')
-                lc = LightCurve(time=time_obj, flux=flux)
-                lc.meta['MISSION'] = attachment['platformId']
-                lc.meta['LABEL'] = attachment['platformId']
-                row_lcs.append(lc)
-            lclist.append(LightCurveCollection(row_lcs))
-        self['lightcurves'] = lclist
+        # if there is no _id column, we can't associate the website data to
+        # the bolides in the BolideDataFrame
+        assert '_id' in self.columns
+
+        # check that ids is not a single string
+        assert type(ids) is not str
+
+        # make ids a List if it is an iterable
+        if hasattr(ids, '__iter__'):
+            ids = list(ids)
+
+            # check that all the given ids actually exist
+            assert all([_id in list(self._id) for _id in ids])
+
+        # if no ids are given, we assume that we will obtain website data for each
+        # event in the BolideDataFrame
+        if ids == None:
+            ids = list(self._id)
```

Comment:
If downloading a lot of data perhaps let the user know this might take a while.

Reply:
There is a tqdm loading bar in `glm_website_event` for when it iterates through events, with a time estimate. Should that be good?

Reply:
Sure, that sound good.